doc/design.md

```diff
@@ -5,55 +5,57 @@ DPPP (DP3, short for Default PreProcessing Pipeline) is a framework
 also create a new MS. When an appropriate input step is in
 place (e.g. a streaming connection to a correlator), MSs can
 be written from scratch.
 
-DPPP supports only regularly shaped MSs, meaning that every
-time slot must have the same number of baselines and antennas.
-Only one spectral window can be used.
+DPPP supports only regularly shaped MSs with the following restrictions:
+- Every time slot has the same number of antennas.
+- Time slots have the same duration and are equally spaced (missing time slots are allowed).
+- Every integration has the same number of correlations and channels.
+- Only one spectral window can be handled.
 
 ## Built-in steps
-DPPP contains built-in steps that are linked together and process one time slot at a time and is therefore suitable for large measurement sets.
-All the built-in steps derive from the [DPStep](@ref DP3::DPPP::DPStep) class.
+DPPP contains built-in steps that are linked together and process one time slot at a time. This way, DPPP is therefore suitable for large measurement sets.
+All the built-in steps derive from the [DPStep](@ref DP3::DPPP::DPStep) class. 
+Its inheritance diagram shows all step types, including the built-in steps.
 
 [DPRun](@ref DP3::DPPP::DPRun) initialises and orchestrates the configured steps.
-It begins with setting the required info for all the first step and its next step(s).
+It begins with setting the required info for the first step and its next step(s).
 Next, each buffer (containing a single time slot) is processed by the first step.
 When processed, it invokes the process function of the next step.
-Once all buffers are processed it finishes the processing of the first step and subsequent steps.
-It wraps it with adding some data to the MeasurementSet written/updated which is started at the last step.
+DPPP calls the `finish` function of the first step and subsequent steps. 
+This function flushes all remaining data from the steps to their output channel(s).
+Finally, DPPP calls the `addToMS` function on each step for updating the metadata of the MS.
 The figure below gives a graphical overview of the [DPStep](@ref DP3::DPPP::DPStep) flow.
 
 ![Process flow of DPPP](doc/images/flow.png)
 
 ### Calibration
 The DPPP step [GainCal](@ref DP3::DPPP::GainCal) implements many variants of direction
-independent calibration. It uses StefCal, with many
+independent calibration. It uses StefCal ([read more](https://ieeexplore.ieee.org/abstract/document/6930038)), with many
 extra features, such as fitting a function over frequency.
 Also full-Jones calibration is supported. Results are written
-to ParmDB; H5Parm support is in development.
+to ParmDB ([read more](https://www.astron.nl/lofarwiki/doku.php?id=public:user_software:documentation:makesourcedb)); H5Parm ([read more](https://github.com/revoltek/losoto/wiki/H5parm-specifications)) support is in development.
```

Comment:
Instead of using 'read more' links, please use StefCal, ParmDB, and H5Parm itself as the link text, instead of 'read more'.